libopenage/gamestate/player.cpp

```diff
@@ -111,5 +114,46 @@ void Player::initialise_unit_types() {
 	}
 }
 
+void Player::active_unit_added(Unit *unit) {
+	// check if unit is actually active
+	if (unit->has_attribute(attr_type::building) && unit->get_attribute<attr_type::building>().completed < 1.0f) {
+		return;
+	}
+
+	// TODO handle here building dependencies
+	// TODO handle here on create unit triggers
+
+	// population
+	if (unit->has_attribute(attr_type::population)) {
+		auto popul = unit->get_attribute<attr_type::population>().population;
+		if (popul < 0) {
+			population.add_capacity(-popul);
+		}
+		else if (popul > 0) {
+			population.add_population(popul);
+		}
+	}
+}
+
+void Player::active_unit_removed(Unit *unit) {
+	// check if unit is actually active
+	if (unit->has_attribute(attr_type::building) && unit->get_attribute<attr_type::building>().completed < 1.0f) {
+		return;
+	}
+
+	// TODO handle here building dependencies
+	// TODO handle here on death unit triggers
+
+	// population
+	if (unit->has_attribute(attr_type::population)) {
+		auto popul = unit->get_attribute<attr_type::population>().population;
+		if (popul < 0) {
+			population.add_capacity(popul);
```

Comment:
to model buildings as negative population is kinda hacky :D 
then again, the only other way would be to have two attributes ("housing space", "housing demand"?).
Maybe we wanna have units that first require an empty population slot, but once created, it provides more living space.